import-export-cli/integration/apiProduct_test.go

```diff
@@ -1126,3 +1127,66 @@ func TestDeleteApiProductWithActiveSubscriptionsSuperTenantUser(t *testing.T) {
 
 	testutils.ValidateAPIProductDeleteFailureWithExistingEnv(t, argsToDelete)
 }
+
+// API products search using query parameters
+func TestApiProductSearchWithQueryParams(t *testing.T) {
+	for _, user := range testCaseUsers {
+		t.Run(user.Description, func(t *testing.T) {
+
+			dev := GetDevClient()
+
+			var searchingApiProductIndex, redundantApiProductIndex int
+			var searchQuery string
+
+			maxIndexOfTheArray := 5
+			minIndexOfTheArray := 0
+
+			// Add the first dependent API to env1
+			dependentAPI1 := testutils.AddAPI(t, dev, user.ApiCreator.Username, user.ApiCreator.Password)
+			testutils.PublishAPI(dev, user.ApiPublisher.Username, user.ApiPublisher.Password, dependentAPI1.ID)
+
+			// Add the second dependent API to env1
+			dependentAPI2 := testutils.AddAPIFromOpenAPIDefinition(t, dev, user.ApiCreator.Username, user.ApiCreator.Password)
+			testutils.PublishAPI(dev, user.ApiPublisher.Username, user.ApiPublisher.Password, dependentAPI2.ID)
+
+			// Map the real name of the API with the API
+			apisList := map[string]*apim.API{
+				"PizzaShackAPI":   dependentAPI1,
+				"SwaggerPetstore": dependentAPI2,
+			}
+
+			// Add set of API Products to env and store api details
+			var addedApiProductsList [numberOfAPIProducts + 1]*apim.APIProduct
+			for apiProductCount := 0; apiProductCount <= numberOfAPIProducts; apiProductCount++ {
+				// Add the API Product to env1
+				apiProduct := testutils.AddAPIProductFromJSON(t, dev, user.ApiPublisher.Username, user.ApiPublisher.Password, apisList)
+				addedApiProductsList[apiProductCount] = apiProduct
+			}
+
+			args := &testutils.ApiProductImportExportTestArgs{
+				CtlUser: testutils.Credentials{Username: user.CtlUser.Username, Password: user.CtlUser.Password},
+				SrcAPIM: dev,
+			}
+
+			//Select random name from the added API Products
+			searchingApiProductIndex = base.GenerateRandomNumber(minIndexOfTheArray, maxIndexOfTheArray)
```

Comment:
Any reason why we are doing this? This will result in a different search taking place for different test runs which means the scenario being tested from time to time is not exactly the same. Its much better to have the same scenario being tested. 

If we need to test different search scenarios, like lower case, upper case, mixed case for example, its better to have different tests for these to ensure we have consistency all the time. But if there are no such scenarios we can simply this to just search for the same thing every time.

Reply:
We are using both context and name as query parameter values. Other than that we can use the same string every time. Will change this according to that.

Reply:
In fact, why dont we just loop through the array and search for all the available APIs that were created(one at a time) and see if the result is the expected match?